[panavdua] iP

panavdua opened the pull request:
# Panav ChatBot
> "Chatbots will be your new best friends." - Christine Crandell ([Forbes](https://www.forbes.com/sites/christinecrandell/?sh=737e00b51ee9))  

Panav frees your mind of having to remember things you need to do and **_doubles_** your productivity. It's,
- text-based
- easy to learn
- **SUPER FAST** to use

All you need to do is, 
1. download it from [here.](https://github.com/panavdua/ip/releases/tag/A-Jar)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 👍 

The best part is that it is completely **FREE**

Features:

- [x] Managing todos, deadlines, and events
- [ ] Managing reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too! Here's the `main` method:
```Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

panavdua committed:
Created Level - 1 framework

panavdua committed:
Level-2 completed

panavdua committed:
Added new Task class and edited Duke according to level 3 increment

panavdua committed:
Added new subclasses of events and completed Level-4

panavdua committed:
Tested out automated testing

panavdua committed:
Did level 5

panavdua committed:
Added delete functionality

panavdua committed:
Fixed bugs in "deadline"

panavdua committed:
Add InvalidNumberException and fix bugs

panavdua committed:
Add javadocs

panavdua committed:
Add functionality to save list of tasks into hard disk

panavdua committed:
Merge branch 'branch-Level-7'

panavdua committed:
Add .gitignore

panavdua committed:
Implement OOP principls

panavdua committed:
Organise all existing classes into packages

panavdua committed:
Merge remote-tracking branch 'origin/add-gradle-support'

panavdua committed:
Add gradle support

panavdua committed:
Configure Gradle to to run JUnit tests

panavdua committed:
Implement JUnit tests for some methods of certain classes

panavdua committed:
Modify details in build.gradle

panavdua committed:
Add JavaDoc comments to all classes and public methods in the program.

panavdua committed:
Made changes to adhere to coding stadard

panavdua committed:
Add feature to find all tasks in the list which contain a specific word

panavdua committed:
Merge branch 'branch-A-JavaDoc'

panavdua committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/panav/command/DeleteCommand.java
#	src/main/java/panav/command/ExitCommand.java
#	src/main/java/panav/command/ListCommand.java
#	src/main/java/panav/exception/DukeException.java
#	src/main/java/panav/exception/InvalidInputException.java
#	src/main/java/panav/exception/InvalidNumberException.java
#	src/main/java/panav/exception/ToDoDescriptionException.java
#	src/main/java/panav/task/Deadline.java
#	src/main/java/panav/task/Event.java
#	src/main/java/panav/task/Task.java
#	src/main/java/panav/task/TaskList.java
#	src/main/java/panav/ui/Ui.java

Resolve conflicts by basic editing

panavdua committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/panav/command/DeadlineCommand.java
#	src/main/java/panav/command/DeleteCommand.java
#	src/main/java/panav/command/EditCommand.java
#	src/main/java/panav/command/EventCommand.java
#	src/main/java/panav/command/ExitCommand.java
#	src/main/java/panav/command/ListCommand.java
#	src/main/java/panav/command/TodoCommand.java

Resolve by basic editing

yanlings commented on src/main/java/panav/command/FindCommand.java:
```diff
@@ -0,0 +1,36 @@
+    String keyWord;
+
+    public FindCommand(String keyWord) {
+        this.keyWord = keyWord;
```
I think keyWord can be written as keyword instead.

yanlings commented on src/main/java/panav/command/TodoCommand.java:
```diff
@@ -0,0 +1,48 @@
+    private String todoMessage;
+
+    public TodoCommand(String todoMessage) {
+        this.todoMessage = todoMessage;
```
Could be toDoMessage because of camelcase.

yanlings reviewed (commented):
Overall, really well done!

panavdua committed:
Implement checkstyle tests and improve the general coding quality

panavdua committed:
Improve checkStyle in various files

panavdua committed:
Add JavaFX basic controls to Panav.java and created Launcher.java

Johnchiahk commented on src/main/java/panav/command/TodoCommand.java:
```diff
@@ -0,0 +1,45 @@
+public class TodoCommand extends Command {
+
+    public static final String COMMAND_WORD = "todo";
+    private String todoMessage;
```
Perhaps toDoMessage may be better?

panavdua replied:
Yup agreed, thanks for the input!

Johnchiahk commented on src/main/java/panav/storage/Storage.java:
```diff
@@ -0,0 +1,100 @@
+            int indexTo = str.indexOf("to:");
+            int check = task.isDone() ? 1 : 0;
+            int length = str.length();
+            if (indexBy != -1) {
+                textToAdd += String.format("%c ~ %d ~ %s ~ %s %n", str.charAt(1), check,
+                        str.substring(7, indexBy - 2), str.substring(indexBy + 4, length - 1));
+            } else if (indexFrom != -1) {
+                textToAdd += String.format("%c ~ %d ~ %s ~ %s ~ %s %n", str.charAt(1), check,
+                        str.substring(7, indexFrom - 1), str.substring(indexFrom + 6, indexTo - 1),
+                        str.substring(indexTo + 4, length - 1));
+            } else {
+                textToAdd += String.format("%c ~ %d ~ %s %n", str.charAt(1), check, str.substring(7));
+            }
```
Great code quality that you have shown, using if, else if, and else 👍

Johnchiahk commented on src/main/java/panav/task/TaskList.java:
```diff
@@ -0,0 +1,101 @@
+    private ArrayList<Task> tasks;
+
+    public TaskList(ArrayList<Task> src) {
+        tasks = new ArrayList<>();
+        for (Task task : src) {
+            tasks.add(task);
+        }
```
Great code quality that you have shown, as the add function in ArrayList is very efficient! :D

Johnchiahk commented on src/main/java/panav/task/ToDo.java:
```diff
@@ -0,0 +1,26 @@
+     * Constructor to initialise attributes.
+     * @param description
+     * @throws ToDoDescriptionException
+     */
+    public ToDo(String description) throws ToDoDescriptionException {
+        super(description);
+        if (description.length() == 0) {
```
Perhaps !description.length() is better?

panavdua replied:
True, shall make the change

Johnchiahk commented on src/main/java/panav/task/Task.java:
```diff
@@ -0,0 +1,51 @@
+     * @return "X" is task is done, " " otherwise.
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
```
Great code quality shown! The choice of conditionals over if-else statements make your code looks cleaner!

Johnchiahk reviewed (commented):
Overall, a very great job Panav :D I hope to write code as well as you do!

panavdua committed:
Create basic layout for GUI Panav

panavdua committed:
Improve code styling and convention

nevinlim commented on src/main/java/panav/command/DeleteCommand.java:
```diff
@@ -0,0 +1,65 @@
+     * @throws InvalidNumberException If the index doesn't exist.
+     */
+    public static int readNumber(String command, int len) throws InvalidNumberException {
+        int number = Integer.parseInt(String.valueOf(command.charAt(command.length() - 1)));
```
Perhaps can split this up into variables instead of nested parentheses to improve code quality

panavdua replied:
Agreed, thanks for the inputs!

nevinlim commented on src/main/java/panav/Panav.java:
```diff
@@ -0,0 +1,76 @@
+                Command c = Parser.parse(fullCommand);
+                c.execute(tasks, ui, storage);
+                if (c.toString().compareTo(ListCommand.COMMAND_WORD) != 0
+                        || c.toString().compareTo(ExitCommand.COMMAND_WORD) != 0) {
```
Maybe can split this into more easily understandable variables

nevinlim commented on src/main/java/panav/command/EditCommand.java:
```diff
@@ -0,0 +1,73 @@
+     * @throws InvalidNumberException If the index doesn't exist.
+     */
+    public static int readNumber(String command, int len) throws InvalidNumberException {
+        int number = Integer.parseInt(String.valueOf(command.charAt(command.length() - 1)));
```
Can reduce this to more readable variables too

panavdua replied:
Thanks for pointing it out :)

nevinlim reviewed (commented):
LGTM, just some minor suggestions to improve code quality 😄

panavdua committed:
Add interaction of GUI with the user

panavdua committed:
Use FXML and SceneBuilder

panavdua committed:
Merge branch 'Level-10'

panavdua committed:
Improve code quality

panavdua committed:
Add assertions to highlight important assumptions in the code

panavdua committed:
Merge pull request #2 from panavdua/branch-A-Assertions

Add assertions to highlight important assumptions in the code

panavdua committed:
Merge branch 'master' of https://github.com/panavdua/ip

panavdua committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/panav/MainWindow.java

panavdua committed:
Update build.gradle

panavdua committed:
Merge pull request #3 from panavdua/branch-A-CodeQuality

Branch a code quality

panavdua committed:
Find command: ability to find all tasks of a particular type

User should have the ability to find all the todos/deadlines/events in
the task list.

Let's add functionality so the user can just type "find todo" to
list out all the todos in the list.

Makes it easier for users to filter out tasks of one type if the
list is too long.

panavdua committed:
Allow case-insensitivity for 'find' command and narrow down on the
'searchable' part of a string representation of a task

panavdua committed:
Merge branch 'branch-C-BetterSearch'

panavdua committed:
Update build.gradle to include JavaFX

panavdua committed:
enable relative path directory instead of absolute so that it can
work on any OS

panavdua committed:
Merge branch 'branch-enableRelativePath'

panavdua committed:
Fix bugs related to 'mark' and 'unmark' command

panavdua committed:
create new classes to separate edit commands

panavdua committed:
create new classes for MarkCommand and UnmarkCommand which extend
EditCommand

panavdua committed:
Add restriction such that user cannot mark tasks which are already
marked; same for unmark as well

panavdua committed:
change panav.txt

panavdua committed:
Merge branch 'branch-separateEditCommands'

panavdua committed:
Add functionality to catch errors arising from editing tasks

panavdua committed:
Create User Guide

panavdua committed:
Update User Guide
